File: public/docs/_examples/dependency-injection/ts/app/app.component.2.ts
// #docregion
// #docregion imports
import { Component }         from '@angular/core';
import { CarComponent }      from './car/car.component';
import { HeroesComponent }   from './heroes/heroes.component.1';

import { Inject }   from '@angular/core';
import { APP_CONFIG, AppConfig,
         HERO_DI_CONFIG }    from './app.config';
import { Logger }            from './logger.service';
// #enddocregion imports

@Component({
  selector: 'my-app',
  template: `
    <h1>{{title}}</h1>
    <my-car></my-car>
    <my-heroes></my-heroes>
  `,
  directives: [CarComponent, HeroesComponent],
  providers: [
    Logger,
   // #docregion providers
    { provide: APP_CONFIG, useValue: HERO_DI_CONFIG }
   // #enddocregion providers
  ]
})
export class AppComponent {
  title: string;

  // #docregion ctor
  constructor(@Inject(APP_CONFIG) config: AppConfig) {
    this.title = config.title;
  }
  // #enddocregion ctor
}

File: public/docs/_examples/dependency-injection/dart/lib/app_component_2.dart
// #docregion

// #docregion imports
import 'package:angular2/core.dart';

import 'app_config.dart';
import 'car/car_component.dart';
import 'heroes/heroes_component_1.dart';
import 'logger_service.dart';
// #enddocregion imports

@Component(
    selector: 'my-app',
    template: '''
      <h1>{{title}}</h1>
      <my-car></my-car>
      <my-heroes></my-heroes>''',
    directives: const [
      CarComponent,
      HeroesComponent
    ],
    providers: const [
      Logger,
      // #docregion providers
      const Provider(APP_CONFIG, useValue: heroDiConfig)
      // #enddocregion providers
    ]
)
class AppComponent {
  final String title;

  // #docregion ctor
  AppComponent(@Inject(APP_CONFIG) Map config)
      : title = config['title'];
  // #enddocregion ctor
}
